src/grt/src/GlobalRouter.cpp

```diff
@@ -635,6 +637,42 @@ void GlobalRouter::findPins(Net* net,
   }
 }
 
+float GlobalRouter::getNetSlack(Net* net)
+{
+  sta::dbNetwork* network = sta_->getDbNetwork();
+  sta::Net* sta_net = network->dbToSta(net->getDbNet());
+  sta::Slack slack = sta_->netSlack(sta_net, sta::MinMax::max());
+  return slack;
+}
+
+void GlobalRouter::computeNetSlacks()
+{
+  // Find the slack for all nets
+  std::unordered_map<Net*, float> net_slack_map;
+  std::vector<float> slacks;
+  for (auto net_itr : db_net_map_) {
+    Net* net = net_itr.second;
+    float slack = getNetSlack(net);
+    net_slack_map[net] = slack;
+    slacks.push_back(slack);
+  }
+  std::stable_sort(slacks.begin(), slacks.end());
+
+  // Find the slack threshold based on the percentage of critical nets
+  // defined by the user
+  int th_index = std::ceil(slacks.size() * critical_nets_percentage_);
```

Comment:
On a design with congestion but positive slack we should not assign any net to have a slack.  I don't think this checks that case.